cloudsim-plus/src/main/java/org/cloudbus/cloudsim/schedulers/cloudlet/CloudletSchedulerAbstract.java

```diff
@@ -528,14 +529,18 @@ public double updateProcessing(final double currentTime, final List<Double> mips
     @SuppressWarnings("ForLoopReplaceableByForEach")
     private double updateCloudletsProcessing(final double currentTime) {
         double nextCloudletFinishTime = Double.MAX_VALUE;
+        long usedPes = 0;
         /* Uses an indexed for to avoid ConcurrentModificationException,
          * e.g., in cases when Cloudlet is cancelled during simulation execution. */
         for (int i = 0; i < cloudletExecList.size(); i++) {
             final CloudletExecution cle = cloudletExecList.get(i);
             updateCloudletProcessingAndPacketsDispatch(cle, currentTime);
             nextCloudletFinishTime = Math.min(nextCloudletFinishTime, cloudletEstimatedFinishTime(cle, currentTime));
+            usedPes += cle.getCloudlet().getNumberOfPes();
         }
 
+        ((VmSimple) vm).setFreePesNumber(vm.getFreePesNumber() - usedPes);
```

Comment:
If we are recomputing the number of free PEs based on the number of PEs from running Cloudlets,
it is `number of existing VM PEs - cloudlets used PEs`, as below.

```suggestion
        ((VmSimple) vm).setFreePesNumber(vm.getNumberOfPes() - usedPes);
```

Reply:
But in the case, when there are new dynamically created cloudlets being scheduled on the `vm`, don't we then need to use the currently free `Pe`s for the above subtraction?

Reply:
If the method were just executed for such new Cloudlets, yes. But it's called several times, while old running Cloudlets and new ones start to execute. Since we are looping over all running Cloudlets, we get the PEs from new and old ones to correctly update the freePesNumber.